rest_framework_simplejwt/views.py

```diff
@@ -28,10 +37,69 @@ def post(self, request, *args, **kwargs):
         except TokenError as e:
             raise InvalidToken(e.args[0])
 
-        return Response(serializer.validated_data, status=status.HTTP_200_OK)
+        response = Response(serializer.validated_data, status=status.HTTP_200_OK)
+
+        if api_settings.AUTH_COOKIE:
+            csrf.get_token(self.request)
+            response = self.set_auth_cookies(response, serializer.validated_data)
+
+        return response
+
+    def set_auth_cookies(self, response, data):
+        return response
 
 
-class TokenObtainPairView(TokenViewBase):
+class TokenRefreshViewBase(TokenViewBase):
+    def extract_token_from_cookie(self, request):
+        return request
+
+    def post(self, request, *args, **kwargs):
+        if api_settings.AUTH_COOKIE:
+            request = self.extract_token_from_cookie(request)
+        return super().post(request, *args, **kwargs)
+
+
+class TokenCookieViewMixin:
+    token_refresh_view_name = 'token_refresh'
+
+    def extract_token_from_cookie(self, request):
+        """Extracts token from cookie and sets it in request.data as it would be sent by the user"""
+        if not request.data:
+            token = request.COOKIES.get('{}_refresh'.format(api_settings.AUTH_COOKIE))
+            if not token:
+                raise NotAuthenticated(detail=_('Refresh cookie not set. Try to authenticate first.'))
+            else:
+                request.data['refresh'] = token
+        return request
+
+    def set_auth_cookies(self, response, data):
+        expires = self.get_refresh_token_expiration()
+        response.set_cookie(
+            api_settings.AUTH_COOKIE, data['access'],
+            expires=expires,
+            domain=api_settings.AUTH_COOKIE_DOMAIN,
+            path=api_settings.AUTH_COOKIE_PATH,
+            secure=api_settings.AUTH_COOKIE_SECURE or None,
+            httponly=True,
+            samesite=api_settings.AUTH_COOKIE_SAMESITE,
+        )
+        if 'refresh' in data:
+            response.set_cookie(
+                '{}_refresh'.format(api_settings.AUTH_COOKIE), data['refresh'],
+                expires=expires,
+                domain=None,
+                path=reverse(self.token_refresh_view_name),
+                secure=api_settings.AUTH_COOKIE_SECURE or None,
+                httponly=True,
+                samesite='Strict',
+            )
+        return response
+
+    def get_refresh_token_expiration(self):
+        return datetime.now() + api_settings.REFRESH_TOKEN_LIFETIME
```

Comment:
Is `datetime.now()` the expected behavior? The server could be anywhere, the client can be anywhere, so perhaps using `django.utils.timezone.now` would be better since it will always use UTC server side. The method `set_cookie` expects a datetime object in UTC: https://docs.djangoproject.com/en/3.0/ref/request-response/#django.http.HttpResponse.set_cookie

Reply:
Idk if L124 would be a problem then.